File: go_echo/model/post.go
package model

import (
    "time"
    "log"
)

type Post struct {
  Id            int64     `json:"id" db:"id,primarykey,autoincrement"`
  TopicId       int64     `json:"topic_id" db:"topic_id,notnull"`
  Name          string    `json:"name" db:"name,notnull`
  Body          string    `json:"body" db:"body,notnull`
  DeletedReason string    `json:"deleted_reason"`
  DeletedAt     time.Time `json:"deleted_at"`
  CreatedAt     time.Time `json:"created_at" db:"created_at"`
  UpdatedAt     time.Time `json:"updated_at" db:"updated_at"`
}

const DISPLAYABLE_SIZE int = 100

func GetLatest(id int) []Post {
    var posts []Post
    dbmap.Select(&posts, "SELECT name, body, created_at from posts WHERE id = ? LIMIT 10", id)
    return posts
}

func GetLatestByTopicId(id int) []Post {
    var posts []Post
    // dbmap.Select(&posts, "SELECT name, body, created_at from posts WHERE topic_id = ? LIMIT ?", id, DISPLAYABLE_SIZE)
    replMap := map[string]interface{}{ "id": id, "limit": DISPLAYABLE_SIZE}
    dbmap.Select(&posts, "SELECT name, body, created_at from posts WHERE topic_id = :id LIMIT :limit", replMap)
    return posts
}

func GetPostsByTopicId(id int, page int) []Post {
    var posts []Post
    // dbmap.SelectOne(&posts, "SELECT * from topics WHERE id = ? LIMIT ?, ?", id, (page - 1) * DISPLAYABLE_SIZE, DISPLAYABLE_SIZE)
    replMap := map[string]interface{}{ "id": id, "offset": (page - 1) * DISPLAYABLE_SIZE, "rowCount": DISPLAYABLE_SIZE }
    dbmap.Select(&posts, "SELECT * from posts WHERE topic_id = :id LIMIT :offset, :rowCount", replMap)
    return posts
}

func DeletePost(id int, deleteReason string) int {
    replMap := map[string]interface{}{ "id": id, "deleteReason": deleteReason }
    result, err := dbmap.Exec("UPDATE posts SET deleted_reason = :deleteReason, deleted_at = cast(NOW() as datetime) WHERE id = :id", replMap)
    if err != nil {
        log.Printf("ERROR: ERROR")
    }
    if c, err := result.RowsAffected(); err != nil {
        return 0
    } else {
        return int(c)
    }
}

func InsertPost(post Post) error {
    replMap := map[string]interface{}{ "topicId": post.TopicId, "name": post.Name, "body": post.Body }
    _, err := dbmap.Exec("INSERT INTO posts (topic_id, name, body) values (:topicId, :name, :body)", replMap)
    return err

    // err := dbmap.Insert(post)
    // return err
}

// NGワードのチェックをしたいけど・・・
func (post Post) NgwordCheck() bool {
    return true
}

File: ts_express/src/model/Post.ts
import DBStore from '../documents/DBStore';

export default class Post {
    private id: number;
    private topicId: number;
    private name: string;
    private body: string;
    private deletedReason: string;
    private deletedAt: string;
    private createdAt: string;
    private updatedAt: string;

    private static DISPLAYABLE_SIZE: number = 100;
    constructor(id: number,
                topicId: number,
                name: string,
                body: string,
                deletedReason: string,
                deletedAt: string,
                createdAt: string,
                updatedAt: string) {

        this.id = id;
        this.topicId = topicId;
        this.name = name;
        this.body = body;
        this.deletedReason = deletedReason;
        this.deletedAt = deletedAt;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public getId(): number {
        return this.id;
    }
    public getTopicId(): number {
        return this.topicId;
    }
    public getName(): string {
        return this.name;
    }
    public getBody(): string {
        return this.body;
    }
    public getDeletedReason(): string {
        return this.deletedReason;
    }
    public getDeletedAt(): string {
        return this.deletedAt;
    }
    public getCreatedAt(): string {
        return this.createdAt;
    }
    public getUpdatedAt(): string {
        return this.updatedAt;
    }

    public static getPostsByTopicId(id: number, page: number): Post[] {
        const posts: Post[] = [];
        try {
            const conn = DBStore.createConnection();
            // プレースホルダーは使えないらしい・・・
            // const sql = 'SELECT * from posts WHERE topic_id = :id LIMIT :offset, :rowCount';
            const sql = 'SELECT * from posts WHERE topic_id = ? LIMIT ?, ?';
            /*
            この書き方はダメっぽい。where条件になっちゃう
            const param: any = {
                id: (id),
                offset: (page - 1) * this.DISPLAYABLE_SIZE,
                rowCount: this.DISPLAYABLE_SIZE,
            };
            */
            const offset = Number((page - 1) * this.DISPLAYABLE_SIZE);
            const rowCount = Number(this.DISPLAYABLE_SIZE);

            // const data = conn.query(sql, param, (err, rows, fields) => {
            const query = conn.query(sql, [id, offset, rowCount], (err, rows, fields) => {
                if (err) { console.log('エラー・・・'); }
                console.log(rows[0]);
                /*
                forEachだとうまくいかない・・・なぜ
                rows.forEach((row) => {
                    // 途中改行の方法がわからないから一旦これで・・・
                    const post = new Post(row.id, row.topic_id, row.name, row.body, row.deleted_reason, row.deleted_at, row.created_at, row.updated_at);
                    posts.push(post);
                });
                */
            });
            console.log(query.sql);
            return posts;
        } catch (err) {
            console.log('エラーだけど・・・');
            return posts;
        }
    }
}
